geomesa-hbase/geomesa-hbase-rpc/src/main/scala/org/locationtech/geomesa/hbase/rpc/coprocessor/GeoMesaHBaseCallBack.scala

```diff
@@ -28,42 +28,37 @@ import org.locationtech.geomesa.utils.index.ByteArrays
  * @param scan initial scan (warning: may be mutated)
  * @param options coprocessor configuration options
  * @param result queue for storing results
- * @param closed indicates if the call has been closed and we should stop querying
  */
-class GeoMesaHBaseCallBack(
-    scan: Scan,
-    options: Map[String, String],
-    result: LinkedBlockingQueue[ByteString],
-    closed: AtomicBoolean
-  ) extends Callback[GeoMesaCoprocessorResponse] with Iterator[Scan] with LazyLogging {
+class GeoMesaHBaseCallBack(scan: Scan, options: Map[String, String], result: LinkedBlockingQueue[ByteString])
+    extends Callback[GeoMesaCoprocessorResponse] with Iterator[Scan] with LazyLogging {
 
   private var updated: Array[Byte] = _ // tracks our status per invocation to detect multiple region updates
 
   val callable: Call[GeoMesaCoprocessorService, GeoMesaCoprocessorResponse] =
     new Call[GeoMesaCoprocessorService, GeoMesaCoprocessorResponse]() {
       override def call(instance: GeoMesaCoprocessorService): GeoMesaCoprocessorResponse = {
-        if (closed.get) { null } else {
```

Comment:
*blinks*  I'm trying to remember why we had the 'closed' variable in the first place. 

You've thought through removing it obviously... right;)?  This was the crux of the matter, right?

Reply:
i think it was from an earlier attempt to get the queries to terminate correctly. i _believe_ it's not needed, but we should verify that it's behaving as expected.